server/jetstream_benchmark_test.go

```diff
@@ -1174,3 +1174,199 @@ func BenchmarkJetStreamKV(b *testing.B) {
 		)
 	}
 }
+
+func BenchmarkJetStreamObjStore(b *testing.B) {
+	const (
+		verbose      = false
+		objStoreName = "B"
+		keyPrefix    = "K_"
+		seed         = 12345
+		initKeys     = true
+
+		// read/write ratios
+		ReadOnly  = 1.0
+		WriteOnly = 0.0
+	)
+
+	// rwRatio to string
+	rwRatioToString := func(rwRatio float64) string {
+		switch rwRatio {
+		case ReadOnly:
+			return "readOnly"
+		case WriteOnly:
+			return "writeOnly"
+		default:
+			return fmt.Sprintf("%0.2f", rwRatio)
+		}
+	}
+
+	// benchmark for object store by performing read/write operations with data of random size
+	RunObjStoreBenchmark := func(b *testing.B, objStore nats.ObjectStore, minObjSz int, maxObjSz int, numKeys int, rwRatio float64) (int, int, int) {
+		var (
+			errors int
+			reads  int
+			writes int
+		)
+
+		rng := rand.New(rand.NewSource(int64(seed)))
+
+		// Each operation is processing a random amount of bytes within a size range which
+		// will be either read from or written to an object store bucket. However, here we are
+		// approximating the size of the processed data with a simple average of the range.
+		b.SetBytes(int64((minObjSz + maxObjSz) / 2))
+
+		for i := 1; i <= b.N; i++ {
+			key := fmt.Sprintf("%s_%d", keyPrefix, rng.Intn(numKeys))
+			var err error
+
+			rwOp := rng.Float64()
+			switch {
+			case rwOp <= rwRatio:
+				// Read Op
+				_, err = objStore.GetBytes(key)
+				reads++
+			case rwOp > rwRatio:
+				// Write Op
+				// dataSz is a random value between min-max object size and cannot be less than 1 byte
+				dataSz := rng.Intn(maxObjSz-minObjSz+1) + minObjSz
+				data := make([]byte, dataSz)
+				rng.Read(data)
+				_, err = objStore.PutBytes(key, data)
+				writes++
+			}
+			if err != nil {
+				errors++
+			}
+
+			if verbose && i%1000 == 0 {
+				b.Logf("Completed: %d reads, %d writes, %d errors. %d/%d total operations have been completed.", reads, writes, errors, i, b.N)
+			}
+		}
+		return errors, reads, writes
+	}
+
+	// benchmark cases table
+	benchmarkCases := []struct {
+		storage  nats.StorageType
+		numKeys  int
+		minObjSz int
+		maxObjSz int
+		rwRatio  float64
+	}{
+		// Memory storage limit: 256MB
```

Comment:
This and the next comment are kinda "out of the blue" they can be wrong at any time, I'd remove them. they are notes for you, not for future readers of this file.